apps/verticalhome/js/pinned_apps_navigation.js

```diff
@@ -13,274 +13,375 @@
+            .element.getAttribute('data-manifesturl') ==
+          app.pinnedAppsNavigation.elements[selectedElemIndex]
+          .getAttribute('data-manifesturl')
+          ){
 
-      if (!this.elements && !this.elements.length) {
-        console.error('Can not find any elements by this selector');
-        return;
+          app.pinnedAppsManager.items[i].index -= 1;
+          app.pinnedAppsManager.items[i-1].index += 1;
+        }
       }
 
-      this.elements[this.selectedElemIndex].classList.add('selected');
-      this.elements[this.selectedElemIndex].classList.add('middle');
-      this.middleElem = true;
+      this.refresh();
     },
 
+    moveItemDown: function moveItemDown() {
 
-    /**
-     * Handle event keydown
-     * @param  {[Object]} e [default object]
-     * @return {[nothing]}   [nothing]
-     */
-    handleEvent: function(e) {
+      this.elemList.classList.add('animation');
 
-      var self = this;
+      var middle = document.querySelector('.middle');
+      if(middle){
+        middle.classList.remove('middle');
+      }
 
-      function cycleElements(index) {
+      var selectedElemIndex = this.currentIndex();
+        if(selectedElemIndex==-1){
+          selectedElemIndex = 1;
 
-        var removed = self.elements.splice(-index, index);
+          app.pinNavigation.elemList.style.top = app.pinNavigation.defTopval;
 
-        for (var i = 0; i < removed.length; i++) {
+          var itm01 = this.elements[this.elements.length-1];
+          var cln01 = itm01.cloneNode(true);
+          this.elemList.insertBefore(cln01, this.elements[0]);
+          itm01.parentNode.removeChild(itm01);
 
-          self.elemList.insertBefore(removed[i], self.elements[i]);
-          self.elements.splice(i, 0, removed[i]);
+          this.elements[selectedElemIndex].classList.add('selected');
 
         }
 
-      }
+      var itm = this.elements[selectedElemIndex];
+      var cln = itm.cloneNode(true);
+      var parentDiv = itm.parentNode;
 
-      if (this.preventDef) {
-        e.preventDefault();
-      }
-
-      function preVerticalNavigation () {
-        if (self.middleElem) {
-          document.querySelector('.middle').classList.remove('middle');
-          self.middleElem = false;
-        }
+      parentDiv.insertBefore(cln, this.elements[selectedElemIndex+2]);
+      itm.parentNode.removeChild(itm);
 
+      var itm0 = this.elements[0];
+      var cln_tmp0 = itm0.cloneNode(true);
+      this.elemList.appendChild(cln_tmp0);
+      itm0.parentNode.removeChild(itm0);
 
-        var toRemoved = self.elemList.querySelector('.removed');
-        var toRemoveOutFocus = self.elemList.querySelector('.out-focus');
+      for(var i=0; i < app.pinnedAppsManager.items.length; i++){
+        if(
+          app.pinnedAppsManager.items[i]
+            .element.getAttribute('data-manifesturl') ==
+          app.pinNavigation
+            .elements[selectedElemIndex].getAttribute('data-manifesturl')
```

Comment:
As I correctly understand you has a variable itm. Is it equal to app.pinNavigation.elements[selectedElemIndex]?